### DISCUSSION TITLE
Budibase and Excel Connection

### DISCUSSION THREAD

**Discussion by DoelliJ** [Budibase/budibase#9910] [Help]
Hi! 

Is there someone who has experience in connecting Budibase with Excel files via a REST API? 
I have several questions e.g. how to set up the API, which authentication is required. 
Furthermore, I am not even sure if the API is the right way how to import data from an excel file because the response is in JSON format. 

Can someone give me a short insight to his/her experiences? 

I would be very thankful!

**Comment by melohagan** [comment]
Hey @DoelliJ 

Is this for an Excel spreadsheet in Office 365? Microsoft do offer an API called Graph: https://learn.microsoft.com/en-us/graph/excel-concept-overview

With that said, if you just want to pull Excel data into a Budibase app and do not need the spreadsheet to be the 'source of truth', then you can import CSV or JSON:

https://docs.budibase.com/docs/csv-import
